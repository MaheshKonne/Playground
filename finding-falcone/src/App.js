import React, {Component, Fragment} from 'react';
import './App.css';
import HeaderComponent from "./Components/HeaderComponent";
import CustomDropdown from "./Components/CustomDropdown";
import FooterComponent from "./Components/FooterComponent";

class App extends Component {
    constructor(props) {
        super(props);
        this.state = {
            name: 'Mahesh Konne',
            planets: []
        }
    }

    getPlanetsAndUpdateState() {
         fetch('https://findfalcone.herokuapp.com/planets').then(response => {
            return response.json();
        }).then(planets => {
            if (planets) {
                this.setState({
                    planets
                })
            }
        }).catch(error => {
            console.log(error);
        });
    }

    componentDidMount() {
       this.getPlanetsAndUpdateState();
    }


    render() {
    return (
        <Fragment>
            <HeaderComponent />
            <div>
                <CustomDropdown title='Destination 1' dropdownValues={['Planet 1', 'Planet 2']} />
            </div>
            <FooterComponent/>
        </Fragment>
    );
  }
}

export default App;