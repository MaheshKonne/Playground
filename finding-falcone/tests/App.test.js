import React from 'react';
import { shallow } from 'enzyme';
import "isomorphic-fetch";

import App from '../src/App';

describe('<App />', () => {
  const component = shallow(<App/>);
  const fetch = jest.fn(() => new Promise(resolve => resolve()));

  it('should match with snapshot', () => {
    expect(component.debug()).toMatchSnapshot();
  });

  it('should call getPlanetsAndUpdateState fetch during componentDidMount', () => {
      const instance = component.instance();
      instance.getPlanetsAndUpdateState();
      expect(component.state().length).not.toBe(0);
  });
})