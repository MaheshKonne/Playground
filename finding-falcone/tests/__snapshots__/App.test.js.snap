// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match with snapshot 1`] = `
"<Fragment>
  <HeaderComponent />
  <div>
    <CustomDropdown title=\\"Destination 1\\" dropdownValues={{...}} />
  </div>
  <FooterComponent />
</Fragment>"
`;
